app/database/admin/data_operator/test.ts

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {MM_TABLES} from '@constants/database';
+import {Q} from '@nozbe/watermelondb';
+import App from '@typings/database/app';
+
+import DatabaseManager, {DatabaseType} from '../database_manager';
+import DataOperator, {IsolatedEntities, OperationType} from './index';
+import {
+    operateAppRecord,
+    operateCustomEmojiRecord,
+    operateGlobalRecord,
+    operateRoleRecord,
+    operateServersRecord,
+    operateSystemRecord,
+} from './operators';
+
+jest.mock('../database_manager');
+
+const {APP, GLOBAL, SERVERS} = MM_TABLES.DEFAULT;
+const {CUSTOM_EMOJI, ROLE, SYSTEM, TERMS_OF_SERVICE} = MM_TABLES.SERVER;
+
+describe('*** Data Operator tests ***', () => {
+    it('=> should return an array of type App for operateAppRecord', async () => {
+        expect.assertions(3);
+
+        const db = await DatabaseManager.getDefaultDatabase();
+        expect(db).toBeTruthy();
+
+        const preparedRecords = await operateAppRecord({
```

Comment:
I think it would be better to exercise `await DataOperator.handleIsolatedEntityData` on the different operators so that we test for the operator assignment and operator functionality at the same time.

Reply:
Done:)